tests/components/lock/test_init.py

```diff
@@ -2,328 +2,261 @@
 
-    await lock.async_handle_lock_service()
-    assert lock.is_locked
-    assert lock.state == STATE_LOCKED
+async def test_lock_states(hass: HomeAssistant, mock_lock: MockLock) -> None:
+    """Test lock entity states."""
 
-    lock._attr_is_unlocking = True
-    assert lock.is_unlocking
-    assert lock.state == STATE_UNLOCKING
+    assert mock_lock.state is None
 
-    await lock.async_handle_unlock_service()
-    assert not lock.is_locked
-    assert lock.state == STATE_UNLOCKED
+    mock_lock._attr_is_locking = True
+    assert mock_lock.is_locking
+    assert mock_lock.state == STATE_LOCKING
 
-    lock._attr_is_jammed = True
-    assert lock.is_jammed
-    assert lock.state == STATE_JAMMED
-    assert not lock.is_locked
+    await help_test_async_lock_service(hass, SERVICE_LOCK)
+    assert mock_lock.is_locked
+    assert mock_lock.state == STATE_LOCKED
 
+    mock_lock._attr_is_unlocking = True
+    assert mock_lock.is_unlocking
+    assert mock_lock.state == STATE_UNLOCKING
 
-async def test_set_default_code_option(
-    hass: HomeAssistant,
-    entity_registry: er.EntityRegistry,
-    enable_custom_integrations: None,
-) -> None:
-    """Test default code stored in the registry."""
-
-    entry = entity_registry.async_get_or_create("lock", "test", "very_unique")
-    await hass.async_block_till_done()
+    await help_test_async_lock_service(hass, SERVICE_UNLOCK)
+    assert not mock_lock.is_locked
+    assert mock_lock.state == STATE_UNLOCKED
 
-    platform = getattr(hass.components, "test.lock")
-    platform.init(empty=True)
-    platform.ENTITIES["lock1"] = platform.MockLock(
-        name="Test",
-        code_format=r"^\d{4}$",
-        supported_features=LockEntityFeature.OPEN,
-        unique_id="very_unique",
-    )
+    mock_lock._attr_is_jammed = True
+    assert mock_lock.is_jammed
+    assert mock_lock.state == STATE_JAMMED
+    assert not mock_lock.is_locked
 
-    assert await async_setup_component(hass, "lock", {"lock": {"platform": "test"}})
-    await hass.async_block_till_done()
 
-    entity0: MockLock = platform.ENTITIES["lock1"]
-    entity_registry.async_update_entity_options(
-        entry.entity_id, "lock", {CONF_DEFAULT_CODE: "1234"}
-    )
-    await hass.async_block_till_done()
+@pytest.mark.parametrize(
+    ("default_code", "code_format", "supported_features"),
+    [("1234", r"^\d{4}$", LockEntityFeature.OPEN)],
+)
+async def test_set_mock_lock_options(
+    hass: HomeAssistant,
+    mock_lock: MockLock,
+) -> None:
+    """Test mock attributes and default code stored in the registry."""
 
-    assert entity0._lock_option_default_code == "1234"
+    assert mock_lock._lock_option_default_code == "1234"
+    state = hass.states.get(TEST_ENTITY_ID)
+    assert state is not None
+    assert state.attributes["code_format"] == r"^\d{4}$"
+    assert state.attributes["supported_features"] == LockEntityFeature.OPEN
 
 
+@pytest.mark.parametrize(("default_code", "code_format"), [("1234", r"^\d{4}$")])
```

Comment:
Since the same `default_code` and `code_format` are used several times, maybe create constants and reuse them? Same comment holds for other repeated values in the other tests.

Reply:
I'm not sure if that will help here, this is not a common practice for tests atm.

Reply:
OK